core/embed/rust/src/ui/model_mercury/component/share_words.rs

```diff
@@ -190,23 +293,67 @@ impl<'a> Component for ShareWords<'a> {
                 self.render_word(self.page_index, target);
             })
         };
-
-        // footer with instructions
-        self.footer.render(target);
     }
 
     #[cfg(feature = "ui_bounds")]
     fn bounds(&self, _sink: &mut dyn FnMut(Rect)) {}
 }
 
 #[cfg(feature = "ui_debug")]
-impl<'a> crate::trace::Trace for ShareWords<'a> {
+impl<'a> crate::trace::Trace for ShareWordsInner<'a> {
     fn trace(&self, t: &mut dyn crate::trace::Tracer) {
-        t.component("ShareWords");
+        t.component("ShareWordsInner");
         let word = &self.share_words[self.page_index as usize];
         let content =
             word.map(|w| build_string!(50, inttostr!(self.page_index as u8 + 1), ". ", w, "\n"));
         t.string("screen_content", content.as_str().into());
         t.int("page_count", self.share_words.len() as i64)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_find_repeated_indices() {
+        let words1 = [
+            TString::from_str("aaa"),
+            TString::from_str("bbb"),
+            TString::from_str("ccc"),
+        ];
+        let words2 = [
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("bbb"),
+        ];
+        let words3 = [
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("bbb"),
+            TString::from_str("bbb"),
+            TString::from_str("aaa"),
+        ];
+        let words4 = [
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+            TString::from_str("aaa"),
+        ];
+
+        assert_eq!(ShareWords::find_repeated(&words1), IndexVec::new());
```

Comment:
Hyper nit: tempted to add a test for empty input array.

Reply:
https://github.com/trezor/trezor-firmware/pull/3954/commits/94c4b7ff7b39a2bf103008c08f60248d17044fee